docs/src/proposals/transactions-v2.md

```diff
@@ -24,76 +24,91 @@ after accounting for signatures and other transaction metadata.
 
 ## Proposed Solution
 
-Introduce a new on-chain program which stores account address maps and add a new
-transaction format which supports concise account references through the
-on-chain address maps.
+1) Introduce a new program which manages on-chain address lookup tables
+2) Add a new transaction format which can make use of on-chain
+address lookup tables to efficiently load more accounts in a single transaction.
 
-### Address Map Program
+### Address Lookup Table Program
 
 Here we describe a program-based solution to the problem, whereby a protocol
 developer or end-user can create collections of related addresses on-chain for
-concise use in a transaction's account inputs. This approach is similar to page
-tables used in operating systems to succinctly map virtual addresses to physical
-memory.
+concise use in a transaction's account inputs.
 
-After addresses are stored on-chain in an address map account, they may be
+After addresses are stored on-chain in an address lookup table account, they may be
 succinctly referenced in a transaction using a 1-byte u8 index rather than a
 full 32-byte address. This will require a new transaction format to make use of
 these succinct references as well as runtime handling for looking up and loading
-accounts from the on-chain mappings.
+accounts from the on-chain lookup tables.
 
 #### State
 
-Address map accounts must be rent-exempt but may be closed with a one epoch
-deactivation period. Address maps must be activated before use.
+Address lookup tables must be rent-exempt when initialized and after
+each time new addresses are appended. Lookup tables can either be extended
+from an on-chain buffered list of addresses or directly by appending
+addresses through instruction data. Newly appended addresses require
+one slot to warmup before being available to transactions for lookups.
```

Comment:
I'm curious, what's the rationale for this? Is there an attack possibility if addresses are immediately available? Or is this because of an implementation detail in how accounts are fetched for address tables?

Reply:
Rationale is here:
```
### Denial of service

Address lookup table accounts may be read very frequently and will therefore
be a more high profile target for denial of service attacks through write locks
similar to sysvar accounts.

For this reason, special handling should be given to address lookup tables.
When an address lookup table is used to lookup addresses for a transaction,
it can be loaded without waiting for a read lock. To avoid race conditions,
only the addresses appended in previous blocks can be used for lookups.
```

Reply:
Gotcha, I thought that was only related to creation, thanks for the clarification